### PR TITLE
Onboarding Replication Policy using autorest

### PR DESCRIPTION
**Pull request by ju993rn4ut**
Get and Remove replicationpolicy commands completed

added default value to subscription id

new policy command

Update policy command and test units

## Description

- I have generated and made changes to all the Get, New, Remove, Update commands in Replication policy using Autorest.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/ChangeLog.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Version 2.0.0
+* Added a feature of remove and update replication policy using a policy object.
+* Fixed the ICreatePolicyInput object to receive Policy specifications using an object.
+* Changed the Input InstanceType to ReplicationScenario for better understanding of the user.
+* Added help examples and test recordings
+  * `Get-AzRecoveryServicesReplicationPolicy`
+  * `New-AzRecoveryServicesReplicationPolicy`
+  * `Remove-AzRecoveryServicesReplicationPolicy`
+  * `Update-AzRecoveryServicesReplicationPolicy`
+* First preview release for module Az.RecoveryServices. The cmdlets to support common scenarios of RecoveryServices like policies, fabrics, protection container, protection container mapping, enable replication and test failover have been exposed.
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/ChangeLog.md]: Remove these changes

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ This directory contains the PowerShell module for the RecoveryServices service.
 This module was primarily generated via [AutoRest](https://github.com/Azure/autorest) using the [PowerShell](https://github.com/Azure/autorest.powershell) extension.
 
 ## Module Requirements
-- [Az.Accounts module](https://www.powershellgallery.com/packages/Az.Accounts/), version 2.7.5 or greater
+- [Az.Accounts module](https://www.powershellgallery.com/packages/Az.Accounts/), version 2.2.3 or greater
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/README.md:20]: Try with 2.7.5 version Az.Accounts itself

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/New-AzRecoveryServicesReplicationPolicy.ps1 [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    )
+
+    process {
+        try {
+            $replicationscenario=$ProviderSpecificInput.ReplicationScenario
+            if($replicationscenario -eq "ReplicateAzureToAzure") {
+                $ProviderSpecificInput.ReplicationScenario = "A2A"
+            }
+            else {
+                throw "For now we just support ReplicateAzureToAzure scenario, working on other scenarios"
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/New-AzRecoveryServicesReplicationPolicy.ps1]: Provided replication scenario is not supported. Only ReplicateAzureToAzure is applicable.

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1 [hunk 4]
```diff
@@ -0,0 +1,82 @@
+function Remove-AzRecoveryServicesReplicationPolicy
+{
+	[OutputType([Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.IJob])]
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1:3]: Indent left

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1 [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    
+        [Parameter(DontShow)]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process
+    {
+        # RsvRef
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1]: Remove

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1 [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        # public string[] ResourceGuardOperationRequest --- this should be a parameter (check in SDK code) ? If yes, optional parameters ? 
+        try {
+            $policyName = $Policy.FriendlyName
+        
+            $null = $PSBoundParameters.Remove("Policy")
+            $null = $PSBoundParameters.Add("PolicyName", $policyName)
+
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Remove-AzRecoveryServicesReplicationPolicy.ps1:73]: Remove this comment

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Update-AzRecoveryServicesReplicationPolicy.ps1 [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Use the default credentials for the proxy
+        ${ProxyUseDefaultCredentials}
+    )
+
+    process {
+        try {
+            $policyName = $Policy.FriendlyName
+            $replicationscenario=$ReplicationProviderSetting.ReplicationScenario
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Update-AzRecoveryServicesReplicationPolicy.ps1]: Align spaces

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Update-AzRecoveryServicesReplicationPolicy.ps1 [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+            $null = $PSBoundParameters.Remove("Policy")
+            $null = $PSBoundParameters.Add("PolicyName", $policyName)
+
+            if($replicationscenario -eq "ReplicateAzureToAzure") {
+                $ReplicationProviderSetting.ReplicationScenario = "A2A"
+            }
+            else {
+                throw "For now we just support ReplicateAzureToAzure scenario, working on other scenarios"
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/custom/Cmdlets/Policy/Update-AzRecoveryServicesReplicationPolicy.ps1]: Change this as specified before

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/docs/Get-AzRecoveryServicesReplicationPolicy.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+Get-AzRecoveryServicesReplicationPolicy -PolicyName <String> -ResourceGroupName <String>
+ -ResourceName <String> [-SubscriptionId <String[]>] [-DefaultProfile <PSObject>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Gets the details of a replication policy.
+
+## EXAMPLES
+
+### Example 1: Get all replication policies in a recovery services vault
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/docs/Get-AzRecoveryServicesReplicationPolicy.md]: List

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/test/Remove-AzRecoveryServicesReplicationPolicy.Tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+      $currentPath = Split-Path -Path $currentPath -Parent
+  }
+  . ($mockingPath | Select-Object -First 1).FullName
+}
+
+Describe 'Remove-AzRecoveryServicesReplicationPolicy' {
+    It 'Delete' {
+        $obj = Get-AzRecoveryServicesReplicationPolicy -PolicyName $env.a2aCreateRemovePolicy -ResourceName $env.a2aVaultName -ResourceGroupName $env.a2aResourceGroupName -SubscriptionId $env.a2aSubscriptionId
+        $obj.Count | Should Not BeNullOrEmpty
+        {Remove-AzRecoveryServicesReplicationPolicy -Policy $obj -ResourceName $env.a2aVaultName -ResourceGroupName $env.a2aResourceGroupName -SubscriptionId $env.a2aSubscriptionId} | Should Not Throw
```
Comment (vidyadharijami) [inline src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/test/Remove-AzRecoveryServicesReplicationPolicy.Tests.ps1:21]: After remove cmdlet, Please add wait for job completion and check that it doesn't exist using Get cmdlet

**Review by vidyadharijami — CHANGES_REQUESTED** [review]
Please address the comments suggested.

File: src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/README.md (hunk 2 above)
Comment (ju993rn4ut) [reply src/RecoveryServices/RecoveryServices.SiteRecovery.Autorest/README.md:20]: But whenever I run autorest, the version depreciates by itself.

**Review by vidyadharijami — APPROVED** [review]

**Pull request merged**